ooni/oonicli.py

```diff
@@ -307,11 +345,163 @@ def post_director_start(_):
         return director.allTestsDone
 
-    def start():
+
+    d.addCallback(setup_nettest)
+    d.addCallback(post_director_start)
+    d.addErrback(director_startup_handled_failures)
+    d.addErrback(director_startup_other_failures)
+    return d
+
+    
+
+# this variant version of runWithDirector splits the process in two,
+# allowing a single director instance to be reused with multiple decks.
+
+def runWithDaemonDirector(logging=True, start_tor=True, check_incoherences=True):
+    """
+    Instance the director, parse command line options and start an ooniprobe
+    test!
+    """
+
+    try:
+        import pika
+        from pika import exceptions
+        from pika.adapters import twisted_connection
+    except ImportError:
+        print "Pika is required for queue connection."
+        print "Install with \"pip install pika\"."
+        sys.exit(7)
+
+
+
+    global_options = setupGlobalOptions(logging, start_tor, check_incoherences)
+
+    director = Director()
+
+    if global_options.get('annotations') is not None:
+        annotations = setupAnnotations(global_options)
+
+    if global_options['no-collector']:
+        log.msg("Not reporting using a collector")
+        global_options['collector'] = None
+        start_tor = False
+    else:
+        start_tor = True
+
+
+    def run_test(global_options, url=None, filename=None):
+        assert url is not None or filename is not None
+
+        deck = createDeck(global_options, url=url, filename=filename)
+
+        d = director.start(start_tor=True,
+                           check_incoherences=check_incoherences)
+
+        def setup_nettest(_):
+            try:
+                return deck.setup()
+            except errors.UnableToLoadDeckInput as error:
+                return defer.failure.Failure(error)
+
+
+
+        # Wait until director has started up (including bootstrapping Tor)
+        # before adding tests
+        def post_director_start(_):
+            for net_test_loader in deck.netTestLoaders:
+                # Decks can specify different collectors
+                # for each net test, so that each NetTest
+                # may be paired with a test_helper and its collector
+                # However, a user can override this behavior by
+                # specifying a collector from the command-line (-c).
+                # If a collector is not specified in the deck, or the
+                # deck is a singleton, the default collector set in
+                # ooniprobe.conf will be used
+
+                collector = setupCollector(global_options, net_test_loader)
+
+                test_details = net_test_loader.testDetails
+                test_details['annotations'] = global_options['annotations']
+
+                director.startNetTest(net_test_loader,
+                                      global_options['reportfile'],
+                                      collector)
+            return director.allTestsDone
```

Comment:
@dantheta also `post_director_start()` could be moved outside of any function and shared.